mm2src/coins/eth.rs

```diff
@@ -3440,6 +3445,31 @@ impl EthCoin {
         Box::new(fut.boxed().compat())
     }
 
+    async fn sign_and_send_transaction_async(
+        &self,
+        value: U256,
+        action: Action,
+        data: Vec<u8>,
+        gas: U256,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let ctx = MmArc::from_weak(&self.ctx)
+            .ok_or("!ctx")
+            .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+        let coin = self.clone();
+        match coin.priv_key_policy {
+            EthPrivKeyPolicy::Iguana(ref key_pair)
+            | EthPrivKeyPolicy::HDWallet {
+                activated_key: ref key_pair,
+                ..
+            } => sign_and_send_transaction_with_keypair(ctx, &coin, key_pair, value, action, data, gas).await,
+            EthPrivKeyPolicy::Trezor => Err(TransactionErr::Plain(ERRL!("Trezor is not supported for EVM yet!"))),
+            #[cfg(target_arch = "wasm32")]
+            EthPrivKeyPolicy::Metamask(_) => {
+                sign_and_send_transaction_with_metamask(coin, value, action, data, gas).await
+            },
+        }
+    }
```

Comment:
I don't think this is needed as you can use the non async version with `compat()`